### PR TITLE
Add model for vyos interfaces resource module

### PR DESCRIPTION
**Pull request by NilashishC**
- This adds a new model for the vyos_interfaces resource module to be consumed by the Resource Module Builder

### CODE REVIEW

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  - preview
+  supported_by:
+  - 'Ansible Network'
+  copyright_str: Copyright 2019 Red Hat
+  license: gpl-3.0.txt
+info:
+  network_os: vyos
+  resource: interfaces
+  version_added: 2.9
+  short_description: 'Manages Interfaces attributes of devices running VyOS.'
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
  short_description: 'Manages interfaces attributes of devices running VyOS'
```

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      items:
+        $ref: '#/definitions/config_dict'
+    state:
+      $ref: '#/definitions/state'
+
+definitions:
+  config_dict:
+    type: object
+    description:
+      - Specifies interfaces related configurations.
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
      - Specifies interfaces related configuration.
```

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    properties:
+      name:
+        description:
+          - Name of the interface, i.e., eth0, eth1.
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
          - Full name of the interface, e.g., eth0, eth1.
```

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        type: str
+        required: true
+      description:
+        description:
+          - Description for the interface.
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
          - Interface description.
```

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        type: str
+      enable:
+        description:
+          - Administrative state of the interface.
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
          - Administrative state of the interface. Set the value to C(true) to administratively enable the interface or  C(false) to disable it.
```

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        type: bool
+        default: true
+      speed:
+        description:
+          - Interface link speed.
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
          - Interface link speed. Applicable for Ethernet interfaces only.
```

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        type: str
+        enum: ['auto', '10', '100', '1000', '2500', '10000']
+      duplex:
+        description:
+          - Interface link duplex mode status.
+        type: str
+        enum: ['full', 'half', 'auto']
+      mtu:
+        description:
+          - Maximum Transmission Unit size for the interface.
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
          - MTU for the interface. 
```

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+          - The value should be between 68 to 9000
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
          - The value should be between 68 to 9000.
```

File: models/vyos/interfaces/merged_example_01.txt [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# Using merged
+
+- name: Configure Ethernet interface
```
Comment (samccann) [inline models/vyos/interfaces/merged_example_01.txt]:
```suggestion
- name: Merge provided configuration with device configuration
```

File: models/vyos/interfaces/overridden_example_01.txt [hunk 10]
```diff
@@ -0,0 +1,14 @@
+
+ # Using overridden
+
+- name: Override interfaces
```
Comment (samccann) [inline models/vyos/interfaces/overridden_example_01.txt]:
```suggestion
- name: Override device configuration of all interfaces with provided configuration
```

File: models/vyos/interfaces/replaced_example_01.txt [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# Using replaced
+
+- name: Configure mentioned interfaces and replace existing config
```
Comment (samccann) [inline models/vyos/interfaces/replaced_example_01.txt]:
```suggestion
- name: Replaces device configuration of listed interfaces with provided configuration
```

**Comment by samccann** [comment]
> Can we add configuration examples for the before and after? (as well as the module output when available)

The examples for eos have this before and after and were very helpful for me to understand what was happening - https://github.com/ansible-network/resource_module_models/pull/7/files

File: models/vyos/interfaces/vyos_interfaces.yaml [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+  - 'Ansible Network'
+  copyright_str: Copyright 2019 Red Hat
+  license: gpl-3.0.txt
+info:
+  network_os: vyos
+  resource: interfaces
+  version_added: 2.9
+  short_description: 'Manages Interfaces attributes of devices running VyOS.'
+  description:
+  - Manages interfaces attributes of devices running VyOS.
```
Comment (samccann) [inline models/vyos/interfaces/vyos_interfaces.yaml]:
```suggestion
  - This module manages the interface attributes of devices running VyOS.
```

**Comment by NilashishC** [comment]
@cidrblock @samccann Addressed the review comments and updated the model to the new format.

**Comment by NilashishC** [comment]
@ikhan2010 @cidrblock Updated the model with VIF support.

**Pull request merged**